esileapclient/tests/functional/base.py

```diff
@@ -0,0 +1,173 @@
+
+        # allow custom configuration to be passed in via env var
+        cfg_file_path = (
+            os.environ.get('OS_ESI_CFG_PATH', '/etc/esi-leap/esi-leap.conf'))
+        cfg_parser = configparser.ConfigParser()
+        if not cfg_parser.read(cfg_file_path):
+            self.fail('Could not open config file %s for reading' %
+                      cfg_file_path)
+
+        auth_opts = {}
+        # main cfg parsing loop
+        try:
+            if cfg_parser.has_section('dummy_node'):
+                opt = 'dummy_node_dir'
+                self.config[opt] = cfg_parser.get('dummy_node', opt)
+            # TODO: add support for keystone v2 (this is v3-only atm)
+
+            # attempts to read authentication credentials in this order:
+            # 1) [keystone] section of config file
+            # 2) [keystone_authtoken] section of config file
+            # 3) from the environment variables (e.g. OS_PASSWORD, etc)
+            for opt in ('username', 'password', 'project_name', 'auth_type',
+                        'auth_url', 'user_domain_name', 'project_domain_name'):
+                for sect in 'keystone', 'keystone_authtoken':
+                    if cfg_parser.has_option(sect, opt):
+                        auth_opts[opt] = cfg_parser.get(sect, opt)
+                        break
+                if opt not in auth_opts.keys():
+                    x = os.environ.get('OS_%s' % opt.upper())
+                    if x is not None:
+                        auth_opts[opt] = x
+                    else:
+                        raise configparser.NoOptionerror
+        except (configparser.NoOptionError):
+            self.fail("Missing option %s in configuration file." % opt)
+
+        self.config['auth_type'] = auth_opts['auth_type']
+        self.config['auth_url'] = auth_opts['auth_url']
+
+        self.users['admin'] = {
+            'name': auth_opts['username'],
+            'password': auth_opts['password'],
+            'domain': auth_opts['user_domain_name']
+        }
+        self.projects['admin'] = {
+            'name': auth_opts['project_name'],
+            'domain': auth_opts['project_domain_name']
+        }
+
+    def _init_client(self, user, project):
+        self.clients[user] = (
+            ESICLIClient(cli_dir=self.config['cli_dir'],
+                         username=self.users[user]['name'],
+                         password=self.users[user]['password'],
+                         user_domain_name=self.users[user]['domain'],
+                         tenant_name=self.projects[project]['name'],
+                         project_domain_name=self.projects[project]['domain'],
+                         identity_api_version='3',
+                         uri=self.config['auth_url']))
+
+    def _init_dummy_project(self, name, roles, parent=None):
+        admin_client = self.clients['admin']
+
+        project_name = data_utils.rand_name('esi-%s-project' % name)
```

Comment:
Could we prefix this with a string that definitively identifies the project as a test project? 'test-esi-%s-project' maybe?